components/RatingStar.js

```diff
@@ -0,0 +1,41 @@
+import styled from "styled-components";
+import { useState } from "react";
+
+const StarRating = ({ rating, onRate }) => {
+  const [hoverRating, setHoverRating] = useState(0);
+
+  return (
+    <Stars>
+      {[1, 2, 3, 4, 5].map((star) => (
+        <Star
+          key={star}
+          filled={star <= (hoverRating || rating)}
+          onClick={() => onRate(star)}
+          onMouseEnter={() => setHoverRating(star)}
+          onMouseLeave={() => setHoverRating(0)}
+        >
+          ★
+        </Star>
+      ))}
+    </Stars>
+  );
+};
+
+export default StarRating;
+
+const Stars = styled.div`
+  display: flex;
+  align-items: center;
+  margin-left: auto;
+`;
+
+const Star = styled.span`
```

Comment:
```suggestion
const Star = styled.li`
```

You will most likely need to change the styling, since a span is an inline-element and an li is a block-element. But having an li is more semantic, since you will change the wrapping element to be a list